pyscript.core/esm/interpreter/micropython.js

```diff
@@ -23,11 +16,18 @@ export default {
         if (config.fetch) await fetchPaths(this, runtime, config.fetch);
         return runtime;
     },
+    setGlobal(interpreter, name, value) {
+        const id = `__pyscript_${this.type}_${name}`;
+        globalThis[id] = value;
+        this.run(interpreter, `from js import ${id};${name}=${id};`);
+    },
+    deleteGlobal(interpreter, name) {
+        const id = `__pyscript_${this.type}_${name}`;
+        this.run(interpreter, `del ${id};del ${name}`);
+        delete globalThis[id];
+    },
```

Comment:
why not:
```js
    setGlobal(interpreter, name, value) {
        const id = `__pyscript_${this.type}_${name}`;
        globalThis[id] = value;
        this.run(interpreter, `from js import ${id} as ${name}`);
        delete globalThis[id];
    },
    deleteGlobal(interpreter, name) {
        this.run(interpreter, `del ${name}`);
    },
```

Reply:
I think I was thinking about the fact the next command could be `runAsync` so I didn't want to break the world if any `await` is involved in that code ... the focus here was more on `setGlobal` and `deleteGlobal` timing (and now I need to check if `runAsync` is consider but IIRC it is) and this code is going to be trashed as soon as MicroPython provides same `globals` API as Pyodide and Wasmoon do already.

Reply:
P.S. now that I think about it yes ... that was a better approach, but like I've said, this code is going to be identical to te Pyodide counter-part hopefully before the TP lands ... if not, this was a really valid hint indeed, thanks for it!

Reply:
On a second thought though ... that doesn't remove the leaked `id` from the Python globals, it just removes it from the *JS* context ... is that OK? I feel like this quick and dirty approach grants both *JS* and global *Python* context get a chance to clean up ... is that so?

Reply:
`from js import id_y as name_z` introduces `globalThis.id_y` as a Python object called `name_z`. It doesn't make anything called `id_y`. The later `del name_z` cleans up the Python variable that was introduced and we immediately removed `id_y` from `globalThis` so everything works out.

Reply:
is that the same in MicroPython as that's the interpreter I am working around lack of better APIs?

Reply:
Should be!